modules/query-performance-analysis.adoc

```diff
@@ -0,0 +1,50 @@
+:_newdoc-version: 2.18.4
+:_template-generated: 2025-09-22
+:_mod-docs-content-type: CONCEPT
+
+[id="query-performance-analysis_{context}"]
+= Query performance analysis
+
+Every query and subquery in Loki generates a `metrics.go` log line with performance statistics. Subqueries emit the log line in the queriers. Every query has an associated single summary `metrics.go` line emitted by the query-front end.
```

Comment:
```suggestion
With Logging 6.4 we introduced new query performance fields to the `Stats` table under the `Summary` section in the Logging view that should help you understand the performance of queries, look for "Time spent in X". If you do not yet see these you can still calculate them since every query and subquery in Loki generates a `metrics.go` log line with performance statistics. Subqueries emit the log line in the queriers. Every query has an associated single summary `metrics.go` line emitted by the query-front end.
```